helpers.go

```diff
@@ -0,0 +1,168 @@
+// schema doesn't define any suitable top-level object.
+func ForceEnvelope(envelopeName string) CUEOption {
+	return func(input *codegen.CueInput) {
+		input.ForcedEnvelope = envelopeName
+	}
+}
+
+// NameFunc specifies the naming strategy used for objects and references.
+// It is called with the value passed to the top level method or function and
+// the path to the entity being parsed.
+func NameFunc(nameFunc simplecue.NameFunc) CUEOption {
+	return func(input *codegen.CueInput) {
+		input.NameFunc = nameFunc
+	}
+}
+
+// TypesFromSchema generates types from a single input schema and a single
+// output language.
+func TypesFromSchema() *SchemaToTypesPipeline {
+	return &SchemaToTypesPipeline{}
+}
+
+// Debug controls whether debug mode is enabled or not.
+// When enabled, more information is included in the generated output,
+// such as an audit trail of applied transformations.
+func (pipeline *SchemaToTypesPipeline) Debug(enabled bool) *SchemaToTypesPipeline {
+	pipeline.debug = enabled
+	return pipeline
+}
+
+// Run executes the codegen pipeline and returns its output.
+func (pipeline *SchemaToTypesPipeline) Run(ctx context.Context) ([]byte, error) {
+	// Validation
+	if pipeline.input == nil {
+		return nil, fmt.Errorf("no input configured")
+	}
+	if pipeline.output == nil {
+		return nil, fmt.Errorf("no output configured")
+	}
+
+	codegenPipeline, err := codegen.NewPipeline()
+	if err != nil {
+		return nil, err
+	}
+	codegenPipeline.Inputs = []*codegen.Input{pipeline.input}
+	codegenPipeline.Transforms = codegen.Transforms{
+		FinalPasses: pipeline.finalPasses,
+	}
+	codegenPipeline.Output = codegen.Output{
+		Types:     true,
+		Languages: []*codegen.OutputLanguage{pipeline.output},
+	}
+
+	// Run the codegen pipeline and return the generated file's content.
+	// Note: since this pipeline is about types with no runtime, we expect a
+	// single file to be generated.
+	generatedFS, err := codegenPipeline.Run(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	generatedFiles := generatedFS.AsFiles()
+	if len(generatedFiles) != 1 {
+		return nil, fmt.Errorf("expected a single generated file, got %d", len(generatedFiles))
+	}
+
+	return generatedFiles[0].Data, nil
+}
+
+/**********
+ * Inputs *
+ **********/
+
+// CUEValue sets the pipeline's input to the given cue value.
+func (pipeline *SchemaToTypesPipeline) CUEValue(pkgName string, value cue.Value, opts ...CUEOption) *SchemaToTypesPipeline {
+	cueInput := &codegen.CueInput{
+		Package: pkgName,
+		Value:   &value,
+	}
+
+	for _, opt := range opts {
+		opt(cueInput)
+	}
+
+	pipeline.input = &codegen.Input{Cue: cueInput}
+
+	return pipeline
+}
+
+/*******************
+ * Transformations *
+ *******************/
+
+// SchemaTransformations adds the given transformations to the set of
+// transformations that will be applied to the input schema.
+func (pipeline *SchemaToTypesPipeline) SchemaTransformations(passes ...compiler.Pass) *SchemaToTypesPipeline {
```

Comment:
Same here, about the `With`
```suggestion
func (pipeline *SchemaToTypesPipeline) WithSchemaTransformations(passes ...compiler.Pass) *SchemaToTypesPipeline {
```